src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -218,6 +218,10 @@ internal static void EnsureNullableAttributeExists(CSharpCompilation compilation
                         {
                             diagnostics.Add(ErrorCode.ERR_BadParameterModifiers, modifier.GetLocation(), SyntaxFacts.GetText(SyntaxKind.RefKeyword), SyntaxFacts.GetText(SyntaxKind.InKeyword));
                         }
+                        else if (parameter.ExclamationToken.Kind() != SyntaxKind.None)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_NullCheckingOnByRefParameter, parameter.ExclamationToken.GetLocation(), parameter);
```

Comment:
>parameter [](start = 130, length = 9)

Error should include parameter name rather than the entire parameter.

Reply:
parameter doesn't have a Name attribute, should I use some kind of ToString instead? Seems wrong

Reply:
`parameter.Identifier.ValueText`

---
In reply to: [303055735](https://github.com/dotnet/roslyn/pull/36979#discussion_r303055735) [](ancestors = 303055735)